common/src/main/java/com/wynntils/core/config/upfixers/ConfigUpfixerManager.java

```diff
@@ -0,0 +1,88 @@
+ */
+package com.wynntils.core.config.upfixers;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.config.upfixers.impl.CustomPoiVisbilityUpfixer;
+import com.wynntils.core.managers.Manager;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ConfigUpfixerManager extends Manager {
+    public final String UPFIXER_JSON_MEMBER_NAME = "upfixers";
+
+    private final List<ConfigUpfixer> configUpfixers = new ArrayList<>();
+
+    public ConfigUpfixerManager() {
+        super(List.of());
+
+        // Register upfixers here, in order of run priority
+        registerUpfixer(new CustomPoiVisbilityUpfixer());
+    }
+
+    public void registerUpfixer(ConfigUpfixer upfixer) {
+        configUpfixers.add(upfixer);
+    }
+
+    /**
+     * Runs all registered upfixers on the given config object.
+     * @param configObject The config object to run upfixers on.
+     */
+    public boolean runUpfixers(JsonObject configObject) {
+        List<ConfigUpfixer> missingUpfixers = getMissingUpfixers(configObject);
+
+        boolean anyChange = false;
+
+        for (ConfigUpfixer upfixer : missingUpfixers) {
+            try {
+                if (upfixer.apply(configObject)) {
+                    anyChange = true;
+                    addUpfixerToConfig(configObject, upfixer);
+                    WynntilsMod.info("Applied upfixer \"" + upfixer.getUpfixerName() + "\" to config.");
+                }
+            } catch (Throwable t) {
+                WynntilsMod.warn("Failed to apply upfixer \"" + upfixer.getUpfixerName() + "\" to config file!", t);
+            }
+        }
+
+        return anyChange;
+    }
+
+    private void addUpfixerToConfig(JsonObject configObject, ConfigUpfixer upfixer) {
+        JsonArray upfixers = configObject.getAsJsonArray(UPFIXER_JSON_MEMBER_NAME);
+
+        if (upfixers == null) {
+            upfixers = new JsonArray();
+            configObject.add(UPFIXER_JSON_MEMBER_NAME, upfixers);
+        }
+
+        upfixers.add(upfixer.getUpfixerName());
+    }
+
+    private List<ConfigUpfixer> getMissingUpfixers(JsonObject configObject) {
+        if (!configObject.has(UPFIXER_JSON_MEMBER_NAME)) return configUpfixers;
+
+        JsonArray upfixers = configObject.getAsJsonArray(UPFIXER_JSON_MEMBER_NAME);
+        if (upfixers == null) return configUpfixers;
+
+        List<String> appliedUpfixers = new ArrayList<>();
+        for (JsonElement upfixer : upfixers) {
+            appliedUpfixers.add(upfixer.getAsString());
+        }
+
+        List<ConfigUpfixer> missingUpfixers = new ArrayList<>();
+
+        for (ConfigUpfixer configUpfixer : configUpfixers) {
+            if (!appliedUpfixers.contains(configUpfixer.getUpfixerName())) {
+                missingUpfixers.add(configUpfixer);
+            }
+        }
```

Comment:
I can't help but think this can be made more compact with streams. Not necessarily better, mark my words, but at least less verbose. :-) Is there not some kind of intersection operation you can use? Your goal is like "get me a list of all upfixers that is not contained in the list of applied upfixers". Sounds like `filter()` to me...